### ISSUE TITLE
EMFILE error upon upload

### ISSUE THREAD

**Issue by ericearl** [OpenNeuroOrg/openneuro#2613]
**Describe the bug**

Upon running an upload command on a folder, I get a WHOLE LOT OF:

```shell
Warning: "some_file.tsv" read zero bytes - check that this file is readable and try again
[Error: EMFILE: too many open files, open 'some_file.tsv'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: 'some_file.tsv'
}
```

Even though the files are readable by me and non-empty.

**To Reproduce**

1. I go to my folder containing the "segregated" dataset folder.
2. I `module load nodejs` then type and execute `~/node_modules/bin/openneuro upload --dataset ds004129 segregated`
3. The stdout prints:

```shell
Adding files to "ds004129"
bids-validator@1.9.3
bids-specification@undefined
        Summary:                 Available Tasks:        Available Modalities:
        10 Files, 10.02MB                                MRI
        1 - Subject
        1 - Session


        If you have any questions, please post on https://neurostars.org/tags/bids.

=======================================================================
13340 files to be uploaded with a total size of 18.5 MB
? Begin upload? (Y/n)
```

4. I type `Y` to begin upload and then this appears:

```
=======================================================================
Starting a new upload (4cee6156) to dataset: 'ds004129'
ds004129 [----------------------------------------] 0% | ETA: 0s | 0/13340[Error: EMFILE: too many open files, open '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/figs.tsv'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/figs.tsv'
}

Warning: "sub-ON87515/phenotype/figs.tsv" read zero bytes - check that this file is readable and try again
[Error: EMFILE: too many open files, open '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/health_history_questions.tsv'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/health_history_questions.tsv'
}
Warning: "sub-ON87515/phenotype/health_history_questions.tsv" read zero bytes - check that this file is readable and try again
[Error: EMFILE: too many open files, open '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/health_rating.tsv'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/health_rating.tsv'
}
Warning: "sub-ON87515/phenotype/health_rating.tsv" read zero bytes - check that this file is readable and try again
[Error: EMFILE: too many open files, open '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/hepatic.tsv'] {
  errno: -24,
  code: 'EMFILE',
  syscall: 'open',
  path: '/home/earlea/data/phenotype/segregated/sub-ON87515/phenotype/hepatic.tsv'
}
# AND MUCH MUCH MORE JUST LIKE THAT
```

6. Even though the files are there and readable and non-zero in size.

**Expected behavior**

I was hoping the data would upload.

**Additional context**

Attempted on a high-throughput networked NIH HPC file server and network transfer task-specific machine.  Two other datasets uploaded just fine from the same exact system, but different folders.

**Comment by nellh** [comment]
How many files are in your local copy of this dataset and can you provide the open file limits for this system?

```shell
ulimit -Hn
ulimit -Hs
```

**Comment by ericearl** [comment]
An addition: after I let it finish failing (about 15 minutes), it also spat this pout before the end (and still no uploaded data):

```shell
Retrying upload for sub-ON87515/phenotype/figs.tsv: Error: Request aborted
Retrying upload for sub-ON87515/phenotype/figs.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/figs.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/figs.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/figs.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: Error: Request aborted
Retrying upload for sub-ON87515/phenotype/health_rating.tsv: Error: Request aborted
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_rating.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_rating.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_rating.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_rating.tsv: ReferenceError: Body already used
Retrying upload for sub-ON87515/phenotype/health_history_questions.tsv: ReferenceError: Body already used
```

**Comment by ericearl** [comment]
@nellh, thanks for checking in!

```shell
Wed May-25 3:09:40pm CPU 14.5%:0:Netwk 9815 earlea@helix:~/data/phenotype
> ulimit -Hn
8192
Wed May-25 3:15:47pm CPU 14.5%:0:Netwk 9860 earlea@helix:~/data/phenotype
> ulimit -Hs
unlimited
Wed May-25 3:15:49pm CPU 14.5%:0:Netwk 9684 earlea@helix:~/data/phenotype
> tree segregated/ | tail
        ├── demographics.tsv
        ├── drug_use.tsv
        ├── ehi.tsv
        ├── eligibility.tsv
        ├── health_history_questions.tsv
        ├── health_rating.tsv
        ├── mental_health_questions.tsv
        └── whodas.tsv

2182 directories, 13339 files
```

So does that mean I need to find an admin to up that number from 8192 to 13339?  Or can I upload it in two halves maybe?  What do you suggest?

**Comment by nellh** [comment]
That's a pretty low hard limit on the number of open files, so that is probably the issue. You shouldn't run into issues with 13k files and 8k limit though, so it is a bug. You could split the dataset into a couple chunks if that doesn't break validation as a workaround.

**Comment by ericearl** [comment]
I split the dataset into two "chunks" to upload it all.  Given this approach, the data do appear to be uploaded okay so far.  Thanks again @nellh !

**Comment by nellh** [comment]
A fix for this has been released in OpenNeuro 4.7.2.
